api/comments/views.py

```diff
@@ -0,0 +1,140 @@
+from modularodm import Q
+from modularodm.exceptions import NoResultsFound
+from rest_framework import generics, permissions as drf_permissions
+from rest_framework.exceptions import NotFound, ValidationError, PermissionDenied
+from api.comments.serializers import CommentSerializer, CommentDetailSerializer, CommentReportsSerializer, CommentReportDetailSerializer, CommentReport
+from api.base.exceptions import Gone
+from api.base import permissions as base_permissions
+from api.nodes.permissions import (
+    ContributorOrPublicForComments
+)
+from framework.auth.oauth_scopes import CoreScopes
+from website.project.model import Comment
+
+
+class CommentMixin(object):
+    """Mixin with convenience methods for retrieving the current comment  based on the
+    current URL. By default, fetches the comment based on the comment_id kwarg.
+    """
+
+    serializer_class = CommentSerializer
+    comment_lookup_url_kwarg = 'comment_id'
+
+    def get_comment(self, check_permissions=True):
+        pk = self.kwargs[self.comment_lookup_url_kwarg]
+        query = Q('_id', 'eq', pk)
+        try:
+            comment = Comment.find_one(query)
+        except NoResultsFound:
+            raise NotFound
+
+        if check_permissions:
+            # May raise a permission denied
+            self.check_object_permissions(self.request, comment)
+        return comment
+
+
+class CommentRepliesList(generics.ListCreateAPIView, CommentMixin):
+    """Replies to a comment.
+
+    By default, a GET will return both deleted and not deleted replies. Comment replies may be
+    filtered by their `deleted` field.
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublicForComments,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_COMMENTS_READ]
+    required_write_scopes = [CoreScopes.NODE_COMMENTS_WRITE]
+
+    serializer_class = CommentSerializer
+
+    def get_queryset(self):
+        return Comment.find(Q('target', 'eq', self.get_comment()))
+
+    # overrides ListCreateAPIView
+    def perform_create(self, serializer):
+        self.check_object_permissions(self.request, self.get_comment())
```

Comment:
There is a flag that allows you to call `check_object_permissions` directly from `get_comment`. This can be simplified accordingly.